packages/kit/src/core/tsconfig.js

```diff
@@ -0,0 +1,101 @@
+import fs from 'fs';
+import path from 'path';
+import { mkdirp } from '../utils/filesystem.js';
+import { SVELTE_KIT } from './constants.js';
+
+/** @param {string} file */
+const exists = (file) => fs.existsSync(file) && file;
+
+/** @param {import('types').ValidatedConfig} config */
+export function generate_tsconfig(config) {
+	const out = path.resolve(SVELTE_KIT, 'tsconfig.json');
+
+	const user_file = exists('tsconfig.json') || exists('jsconfig.json');
+
+	const paths = {};
+
+	paths['$lib'] = [path.relative(SVELTE_KIT, config.kit.files.lib)];
+	paths['$lib/*'] = [path.relative(SVELTE_KIT, config.kit.files.lib) + '/*'];
+
+	if (user_file) {
+		// we have to eval the file, since it's not parseable as JSON (contains comments)
+		const user_tsconfig_json = fs.readFileSync(user_file, 'utf-8');
+		const user_tsconfig = (0, eval)(`(${user_tsconfig_json})`);
```

Comment:
Eesh, this makes me extremely uncomfortable. I don't know what else I would suggest. Regexes to strip the comments (somehow) and then parse it?

Reply:
A very quick search reveals https://stackoverflow.com/a/62945875 which suggests `str.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, (m, g) => g ? "" : m)` which I have not tested at all.

Reply:
I do realize that, yes, any code in your app's directory is stuff you're supposed to trust, and that this is probably an area that it's not worth trying to protect people from themselves, but still. This gives me the creeps.

Reply:
yeah, i didn't think an `eval` would get past the Conduitry filter, even if it's made indirect with `(0, eval)`. What's the danger though? Presumably an attacker capable of modifying `tsconfig.json` could just as equally modify `src/hooks.js`, which has far more potential for mischief

Reply:
Yeah, I suppose that, rationally, there's not any great harm in this, nor any great benefit in avoiding this one particular attack vector. I'll resolve this conversation, but I got my eye on you Mr. Harris.